accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,514 @@
+    - Rollback/fast-forward attacks (e.g. providing the wrong version for update)
+    - Freeze/endless data attacks (e.g. preventing updates)
+    - Extraneous dependencies attacks (e.g. tricking the client to install undeclared dependencies)
+  - Malicious maintainer (e.g. [event-stream](https://blog.npmjs.org/post/180565383195/details-about-the-event-stream-incident))
+  - Compromised login credentials (where 2FA can help)
+  - Typosquatting attacks (e.g. `lodahs`), [“dependency confusion”](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610)
+  - Accidental vulnerabilities
+
+## Sigstore as signing solution
+The [Sigstore](https://www.sigstore.dev/) project has been selected as the solution to signing npm packages (see detailed explanation below). It is currently the only working solution that supports our key requirements:
+- Links packages hosted on the npm registry to the source and build they originated from (provenance information).
+- Supports signing identities that don't expose any personally identifiable information (PII) about maintainers, e.g. emails.
+- Avoids developer-managed keys (as there's no good way to offer trust to the community given the challenges of distributing public keys).
+
+Note that nothing in this current proposal precludes us from offering additional signing methods in the future as they become available and mature.
+
+## Risks of adopting Sigstore
+Sigstore infrastructure and tooling are currently not considered production ready and are run on a best-effort basis. The Sigstore project is working towards a [General Availability](https://blog.sigstore.dev/an-update-on-general-availability-5c5563d4e400) release later this year but we don't yet know what these guarantees will look like yet.
+
+As such, there are several risks of adopting Sigstore for npm that are worth calling out:
+- No buy-in from the broader open source npm community.
+- Sigstore's public and free-to-use infrastructure is not able to meet our uptime requirements (>99.9%).
+- Sigstore is not able to secure the funding required to run the infrastructure with production-grade support, SLOs and on-call rotation.
+
+To mitigate some of these risks, GitHub is planning to work directly with Sigstore to define and support robust production-grade SLOs for uptime and reliability that meet the requirements for npm. This will involve being on-call and wearing the pager for production services.
+
+In addition, GitHub is planning to support Sigstore by:
+- Running the signed timestamp authority for Rekor as well as a Certificate Transparency monitor for Fulcio. This will help spread the trust outside the four walls of Sigstore.
+- Maintaining the Sigstore trust root and having a root key holder as part of the group of Sigstore root key holders.
+
+We also remain open to an alternative solution if Sigstore is not able to meet npm's requirements for uptime and support.
```

Comment:
Are there any potential alternatives that have been identified if this is the case?

Reply:
Not concrete alternatives but the rough shape would probably look similar to the architecture in this RFC with the difference that npm/github would be running the services instead. Either fully fledged or a paired down version.

It would be sad if we got to this as it would probably mean that we either end up with some proprietary signature service just for npm/github or potentially multiple competing public good instances.

Reply:
Another alternative with minimal impact to the rest of this RFC is npm running or funding its own instance of Sigstore's software stack, primarily rekor.